admiral/pkg/clusters/virtualservice_routing.go

```diff
@@ -883,21 +905,61 @@ func getDestinationsForGTPDNSPrefixes(
 			continue
 		}
 
+		var newDNSPrefixedSNIHost, routeHost string
+
 		for _, policy := range globalTrafficPolicy.Spec.Policy {
-			if policy.DnsPrefix == common.Default || policy.DnsPrefix == env {
+			weights := make(map[string]int32)
+			var remoteRD *vsrouting.RouteDestination
+			if policy.Target != nil {
+				for _, target := range policy.Target {
+					weights[target.Region] = target.Weight
+				}
+			}
+
+			if !updateWeights && len(weights) == 0 && (policy.DnsPrefix == common.Default || policy.DnsPrefix == env) {
 				continue
+			} else if policy.DnsPrefix == common.Default || policy.DnsPrefix == env {
+				routeHost = hostWithoutSNIPrefix
+			} else {
+				routeHost = policy.DnsPrefix + common.Sep + hostWithoutSNIPrefix
 			}
-			newDNSPrefixedSNIHost, err := generateSNIHost(policy.DnsPrefix + common.Sep + hostWithoutSNIPrefix)
+
+			newDNSPrefixedSNIHost, err = generateSNIHost(routeHost)
 			if err != nil {
 				return nil, err
 			}
+
 			newRD, err := copyRouteDestinations(routeDestinations)
 			if err != nil {
 				return nil, err
 			}
+
+			if len(weights) > 0 && updateWeights {
+				for _, rd := range newRD {
+					if strings.HasSuffix(rd.Destination.Host, "svc.cluster.local") {
+						if weights[sourceClusterLocality] == 0 {
+							rd.Destination.Host = routeHost
+							rd.Destination.Port = &networkingV1Alpha3.PortSelector{
+								Number: 80,
+							}
+						} else if weights[sourceClusterLocality] != 100 {
+							if rd.Weight != 0 {
+								weight := (float32(rd.Weight) / 100) * float32(weights[sourceClusterLocality])
+								rd.Weight = int32(weight)
+							} else {
+								rd.Weight = weights[sourceClusterLocality]
+							}
+							remoteRD = getRouteDestination(routeHost, 80, 100-weights[sourceClusterLocality])
```

Comment:
hey @vrushalijoshi , sorry if I misunderstood this. In this case, if the `sourceClusterLocality=west` and its weight is set to 20%, that means that 20% of the traffic should go to `.local`. But it seems it is doing the opposite. Correct me if I'm wrong though.

Reply:
Hey, it will update the 20% traffic to route destination (rd) which is basically updating the ".local" endpoint( in this case west). For east a new endpoint with ".global" would be created with 80%

Reply:
oh sorry it makes sense. I missed that